### ISSUE TITLE
Toggle questions do not update activity counter 

### ISSUE THREAD

**Issue by bhoffman0** [RunestoneInteractive/RunestoneComponents#1246]
@vqum the toggle questions are not updating the activity counter at the bottom of the page which is still stuck at 1 after completing them. See https://runestone.academy/runestone/books/published/py4e-int/functions/functions_mixedupcode.html , do problem 1 and check activity counter at bottom:
![image](https://user-images.githubusercontent.com/4571962/130118671-28880561-6c10-41b6-9c66-73daa345e61f.png)

**Comment by Ebenezer997** [comment]
@bhoffman0 ”Has this issue been resolved?”

**Comment by bhoffman0** [comment]
No it's does not work.

**Comment by vqum** [comment]
I am looking into this issue

**Comment by bnmnetp** [comment]
The problem for the activity counting and the toggle questions comes from the following:

The server (see books.py) queries the database to get a list of all of the components on the page, using the chapter, subchapter, and basecourse.   This data comes from the questions table in the database.  This list is transformed to a dictionary with a value of 0 or 1 to indicate whether the user has interacted with the component at some point in the past. 

This list is embedded in the eBookConfig object as part of the page when it is served:

```javascript
eBookConfig.activities = {"select_u1_muc_wc14": 0, "select_u1_muc_wc13": 0, "select_u1_muc_wc12": 0,
 "select_u1_muc_wc10": 0, "select_u1_muc_wc9": 0, "select_u1_muc_wc8": 0, "select_u1_muc_wc6": 0,
 "select_u1_muc_wc3": 0, "select_u1_muc_wc2": 0, "select_u1_muc_wc7": 0, "select_u1_muc_wc5": 0, 
"select_u1_muc_wc4": 0, "select_u1_muc_wc11": 0, "select_u1_muc_wc1": 0, 
"1. Getting Started and Primitive Types/1.9 Experimental: Choose Mixed Up or Write Code Practice": 0}
```

Notice that the keys are the unique identifiers for the selectquestion.   None of the **actual** questions are in this data structure.

when the student interacts with one of the **actual** questions the code attempts to update this dictionary, using activities[div_id]++ the keys is added to the dictionary but the value ends up as `NaN`. So the total never increases.

### We have multiple issues

1. The **actual** questions for 1.9 are in the database but their chapter is `HiddenFiles` and their subchapter is `topic-1-8-toggle-write-code`  So, they will never get selected and added to the data structure by the server.
2. Because the code assumes that the key will be in the dictionary we end up with a `NaN`
3. It is not clear how we **should** be counting the activities.  If a student does both the parsons and the activecode should that still only count as one thing?


### Solution idea 1

1. Make use of the `:chapter:` and `:subchapter:` options on the actual questions so that they match the chapter and subchapter that contains the selectquestion.  This way they will be included in the list and we'll be able to pick up on the historical interactions with them.
2. Mark the select questions as `:optional:` so that they are not included in the list. -- I think this is the right plan because we are never going to have an entry in the database for activity on the selectquestion.
2. If we just do that then we will at least show **some** progress, but unless a student does both options for every question they will never get to 100% for a page like 1.9

### Solution idea 2

1. Don't do any of the things in idea 1
2. Each **actual** question that is part of a toggle questions knows the id of the selectquestion. if `this.selector_id` is not undefined then we know this question is part of a `selectquestion` / toggle question. If this attribute is present then pass selector_id to the updateProgress method (see runestonebase.js)
3. If that is all we do that will count 1 for every toggle question on the page. and it will seem seem correct to the student (I think).  But we will also need to add an event to the log for the selectquestion as well otherwise when the page is reloaded the counts for all the toggle questions will reset to 0.
4. (alternate 3) The other possible solution to the second half of 3 is to make the logic on the server more complicated and have it detect that the question is a `selectquestion` and then go looking for the associated actual question (if it is there).  I'm not a fan of this as I want to keep the logic for serving pages as simple and fast as possible.


@barbarer @bhoffman0 @vqum What do you all think??

**Comment by barbarer** [comment]
For the :toggle: I want the ebook to score whichever question the user
selected.  I do want them to get points for doing the toggle questions.

Doesn't an assignment know all the questions in the assignment?  If so
can't we just determine which question in a select question was actually
selected and score that?

Dr. Barbara Ericson
Assistant Professor, School of Information
University of Michigan

**Comment by bnmnetp** [comment]
Yes, that is what the autograder will do, this issue is about the progress bar at the bottom of the page.

**Comment by barbarer** [comment]
Ah, then I lean toward solution 2.  I want the select question to count as
one activity.

Dr. Barbara Ericson
Assistant Professor, School of Information
University of Michigan

**Comment by bnmnetp** [comment]
If @bhoffman0 agrees, then we should implement solution 2.  @vqum do you think you can get it done by Friday or should I go ahead and do this?  I would really like to have it in the build so that it is working for CSAwesome by next week.

**Comment by bhoffman0** [comment]
Sounds good to me.  Thanks for working on this!

**Comment by vqum** [comment]
@bnmnetp I'm still encountering issues with building local Runestone server on my machine. I could try to tackle this, but any code I produce would be written blind since I have no way to test it locally.

**Comment by bnmnetp** [comment]
OK, I don't think it will take me very long at all to do this, since I've looked at all the code just today trying to figure out what was going wrong.

I'm sorry you are having trouble.  I've just done two machines from scratch in the last two days getting prepared to test the new peer instruction features.  I really don't know what the problem could be for you.

**Comment by bhoffman0** [comment]
Wow awesome! Thank you!

**Comment by barbarer** [comment]
Terrific!

Dr. Barbara Ericson
Assistant Professor, School of Information
University of Michigan
